### ISSUE TITLE
Implement new Concept Exercise: loops

### ISSUE THREAD

**Issue by workingjubilee** [exercism/rust#1062]
This issue describes how to implement the loops concept exercise for the Rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should convey a basic understanding of loops in Rust. A complete review of all possible loops in a single exercise is infeasible. The primary goal is to introduce the student to the usage of `for` loops with the `x..y` pattern, which are the most common type of loop in Rust.

## Learning objectives

Student should be able to...
- use a `for` loop
- understand the application of the `1..=x` range pattern in a `for` loop

Student may be able to...
- use a conditional `while` loop
- use an infinite `loop` with a `break` condition

## Out of scope

- more advanced control flow like `while let` loops
- explicit usage of iterators (even though this is an implicit usage of an iterator!)

## Concepts

- loops
- ranges
- patterns

## Prerequisites

This exercise should depend only on a basic understanding of functions and variables.

### Hints

- For: https://doc.rust-lang.org/std/keyword.for.html

### After

- Rust by Example (For and Range): https://doc.rust-lang.org/rust-by-example/flow_control/for.html
- The Rust Programming Language (Patterns and Loops): https://doc.rust-lang.org/book/ch18-01-all-the-places-for-patterns.html
- Rust Control Flow: https://doc.rust-lang.org/book/ch03-05-control-flow.html

## Representer

This should require no special changes to the representer.

## Analyzer

This should require no special changes to the analyzer.

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance. In addition you may want to make use of the shell script while in the `concept` directory, with the arg loops, e.g. `sh boil_exercise.sh loops`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by senekor** [comment]
After several failed attempts to create a high-quality syllabus, I believe any future attempts will have to do their own design work from scratch.
